### ISSUE TITLE
django-tastypie for 0.17 cannot be a dependency link

### ISSUE THREAD

**Issue by benjaoming** [learningequality/ka-lite#5319]
## Summary

Seems that some old version of pip will mis-behave, and the dependency links are scheduled to be removed in Pip 9.
## System information
- Version: 0.17
## Traceback or relevant snippet from server.log

```

pip install ka-lite --pre

Collecting ka-lite

  Downloading ka_lite-0.17.0b1-py2-none-any.whl (54.7MB)

    100% |████████████████████████████████| 54.7MB 19kB/s 

Collecting cherrypy==3.2.2 (from ka-lite)

  Downloading CherryPy-3.2.2.zip (464kB)

    100% |████████████████████████████████| 471kB 969kB/s 

Collecting django-tastypie==0.12.2dev0 (from ka-lite)

  Could not find a version that satisfies the requirement django-tastypie==0.12.2dev0 (from ka-lite) (from versions: 0.9.6, 0.9.7, 0.9.9, 0.9.10, 0.9.11, 0.9.12, 0.9.13, 0.9.14, 0.9.15, 0.9.16, 0.10.0, 0.11.0, 0.11.1, 0.12.0, 0.12.1, 0.12.2, 0.13.0, 0.13.1, 0.13.2, 0.13.3)

No matching distribution found for django-tastypie==0.12.2dev0 (from ka-lite)
```
## How to reproduce

Happens on OSX

**Comment by radinamatic** [comment]
@benjaoming 
Hmmm, strange because I already had the 0.17 installed previously on that Ubuntu machine, and I can't remember it gave me the same issue... :thinking:

**Comment by benjaoming** [comment]
You'll only see the error on pip 9 + `ka-lite` (not `ka-lite-static`).

**Comment by radinamatic** [comment]
pip version 8.1.2 seems to fail too...
```
ubmooc@ubmooc-VirtualBox:~$ pip --version
pip 8.1.2 from /usr/lib/python2.7/dist-packages (python 2.7)
ubmooc@ubmooc-VirtualBox:~$ pip install ka-lite --pre
Collecting ka-lite
  Downloading ka_lite-0.17.0b3-py2-none-any.whl (32.0MB)
    100% |████████████████████████████████| 32.0MB 54kB/s 
Collecting requests==2.11.1 (from ka-lite)
  Downloading requests-2.11.1-py2.py3-none-any.whl (514kB)
    100% |████████████████████████████████| 522kB 2.0MB/s 
Collecting cherrypy==3.3.0 (from ka-lite)
  Downloading CherryPy-3.3.0.tar.gz (429kB)
    100% |████████████████████████████████| 430kB 3.1MB/s 
Collecting youtube-dl==2015.11.24 (from ka-lite)
  Downloading youtube_dl-2015.11.24-py2.py3-none-any.whl (1.1MB)
    100% |████████████████████████████████| 1.1MB 1.0MB/s 
Collecting django<1.6 (from ka-lite)
  Downloading Django-1.5.12-py2.py3-none-any.whl (8.3MB)
    100% |████████████████████████████████| 8.3MB 180kB/s 
Collecting South==1.0.2 (from ka-lite)
  Downloading South-1.0.2.tar.gz (96kB)
    100% |████████████████████████████████| 102kB 7.0MB/s 
Collecting python-dateutil==2.4.2 (from ka-lite)
  Downloading python_dateutil-2.4.2-py2.py3-none-any.whl (188kB)
    100% |████████████████████████████████| 194kB 4.1MB/s 
Collecting docopt<0.7,>=0.6 (from ka-lite)
  Downloading docopt-0.6.2.tar.gz
Collecting ply==3.4 (from ka-lite)
  Downloading ply-3.4.tar.gz (138kB)
    100% |████████████████████████████████| 143kB 5.9MB/s 
Collecting rsa==3.1.1 (from ka-lite)
  Downloading rsa-3.1.1.tar.gz
Collecting six==1.9.0 (from ka-lite)
  Downloading six-1.9.0-py2.py3-none-any.whl
Collecting ifcfg==0.9.3 (from ka-lite)
  Downloading ifcfg-0.9.3-py2-none-any.whl
Collecting importlib==1.0.3 (from ka-lite)
  Downloading importlib-1.0.3.tar.gz
Collecting slimit==0.8.1 (from ka-lite)
  Downloading slimit-0.8.1.zip (88kB)
    100% |████████████████████████████████| 92kB 8.3MB/s 
Collecting django-appconf==1.0.1 (from ka-lite)
  Downloading django_appconf-1.0.1-py2.py3-none-any.whl
Collecting django-annoying==0.8.4 (from ka-lite)
  Downloading django-annoying-0.8.4.tar.gz
Collecting pbkdf2==1.3 (from ka-lite)
  Downloading pbkdf2-1.3.tar.gz
Collecting django-js-reverse==0.5.0 (from ka-lite)
  Downloading django_js_reverse-0.5.0-py2.py3-none-any.whl
Collecting peewee==2.6.4 (from ka-lite)
  Downloading peewee-2.6.4.tar.gz (464kB)
    100% |████████████████████████████████| 471kB 2.0MB/s 
Collecting django-tastypie==0.12.2dev0 (from ka-lite)
  Could not find a version that satisfies the requirement django-tastypie==0.12.2dev0 (from ka-lite) (from versions: 0.9.6, 0.9.7, 0.9.9, 0.9.10, 0.9.11, 0.9.12, 0.9.13, 0.9.14, 0.9.15, 0.9.16, 0.10.0, 0.11.0, 0.11.1, 0.12.0, 0.12.1, 0.12.2, 0.13.0, 0.13.1, 0.13.2, 0.13.3)
No matching distribution found for django-tastypie==0.12.2dev0 (from ka-lite)
You are using pip version 8.1.2, however version 9.0.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
ubmooc@ubmooc-VirtualBox:~$ 
```

**Comment by benjaoming** [comment]
@radinamatic that's odd... but doesn't matter, sorry that the description of the symptoms hasn't been accurate enough, but I've only encountered the problem as #5353. Fix remains the same.

The fix is that I've had to create a patched fork of django-tastypie, I called it django-tastypie-legacy.

**Comment by benjaoming** [comment]
Fixed in #5373
